### PR TITLE
add name modifier

### PR DESCRIPTION
**Pull request by frederikhors**

### CODE REVIEW

**Comment by frederikhors** [comment]
@deankarn are you interested in merging these new PRs for new modifiers?

**If you don't have time can you please give me permission for the merge?**

I would like to use this package in my current projects and since I have done a lot of PR in many different branches I would like to use only the original and avoid headaches.

I can help you out here.

Thanks.

**Comment by deankarn** [comment]
Thanks @frederikhors 

I am most definitely interested in merging and will be trying to review this week. Sorry for the delay, the family and I just moved into our new house and so that has been the priority over the last couple weeks.

**Comment by frederikhors** [comment]
Please don't forget about me. ♥

File: modifiers/string.go [hunk 1]
```diff
@@ -89,6 +91,35 @@ func SnakeCase(ctx context.Context, t *mold.Transformer, v reflect.Value, param
...
+		return nil
+	}
+	first := onlyOne(strings.ToLower(s), []map[string]string{
+		{`[^\pL-\s']`: ""}, // cut off everything except [ alpha, hyphen, whitespace, apostrophe]
+		{`\s{2,}`: " "},    // trim more than two whitespaces to one
+		{`-{2,}`: "-"},     // trim more than two hyphens to one
+		{`'{2,}`: "'"},     // trim more than two apostrophes to one
+		{`( )*-( )*`: "-"}, // trim enclosing whitespaces around hyphen
+	})
+	v.SetString(strings.Title(regexp.MustCompile(`[\p{L}]([\p{L}|[:space:]|\-|\']*[\p{L}])*`).FindString(first)))
```
Comment (deankarn) [inline modifiers/string.go:110]: compiling regexes every invocation is costly, can these be moved to global variables for reuse?

**Review by deankarn — CHANGES_REQUESTED** [review]

File: modifiers/string.go (hunk 1 above)
Comment (frederikhors) [reply modifiers/string.go]: I agree. Done.

File: modifiers/string.go [hunk 2]
```diff
@@ -99,6 +100,37 @@ func TitleCase(ctx context.Context, t *mold.Transformer, v reflect.Value, param
...
+// converts multiple spaces and dashes to single characters, title cases multiple names.
+// Example: "3493€848Jo-$%£@Ann " -> "Jo-Ann", " ~~ The Dude ~~" -> "The Dude", "**susan**" -> "Susan",
+// " hugh fearnley-whittingstall" -> "Hugh Fearnley-Whittingstall"
+func NameCase(ctx context.Context, t *mold.Transformer, v reflect.Value, param string) error {
+	s, ok := v.Interface().(string)
+	if !ok {
+		return nil
+	}
+	first := onlyOne(strings.ToLower(s), namePatterns)
+	v.SetString(strings.Title(regexp.MustCompile(`[\p{L}]([\p{L}|[:space:]\-']*[\p{L}])*`).FindString(first)))
```
Comment (deankarn) [inline modifiers/string.go:121]: Fi the regex can be made into a global that would be more efficient.

Comment (frederikhors) [reply modifiers/string.go:121]: Ok. I'll do it, but now I'm asking if there is a better way to do this, maybe without regex? What do you think?

Comment (frederikhors) [reply modifiers/string.go]: Done.

Comment (deankarn) [reply modifiers/string.go]: I think it'd be more efficient without the regex, but regex works for the first iteration and con be optimized later.

**Pull request merged**